### PR TITLE
Bumps go version to 1.21 (latest stable)

### PR DESCRIPTION
**Pull request by gatici**
This PR bumps the go version to 1.21 which is the latest stable release. 

The update is accomplished by running following commands:
```
go mod edit -go=1.21
go mod tidy
```

Reference: https://tip.golang.org/doc/go1.21

### CODE REVIEW

**Comment by gatici** [comment]
This PR is tested by building new image successfully.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/omec-project/sctplb
 
-go 1.17
+go 1.21
```
Comment (thakurajayL) [inline go.mod:3]: did you run go mod tidy after this change ?

Comment (gatici) [reply go.mod:3]: Yes. I run `go mod tidy` and docker build is also completed successfully.

```
$ go mod tidy
go: downloading google.golang.org/protobuf v1.28.0
go: downloading golang.org/x/sys v0.0.0-20211216021012-1d35b9e2eb4e
go: downloading gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127
go: downloading github.com/stretchr/testify v1.7.0
go: downloading golang.org/x/net v0.0.0-20211112202133-69e39bad7dc2
go: downloading github.com/google/go-cmp v0.5.6
go: downloading google.golang.org/genproto v0.0.0-20200825200019-8632dd797987
go: downloading github.com/kr/pretty v0.1.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading github.com/kr/text v0.1.0

$ make docker-build
for target in sctplb; do \
	DOCKER_BUILDKIT=1 docker build   \
		--target $target \
		--tag $target:0.0.1-dev \
		--build-arg org_label_schema_version="0.0.1-dev" \
		--build-arg org_label_schema_vcs_url="" \
		--build-arg org_label_schema_vcs_ref="" \
		--build-arg org_label_schema_build_date="2023-11-14T08:12:22Z" \
		--build-arg org_opencord_vcs_commit_date="" \
		. \
		|| exit 1; \
done
[+] Building 34.4s (15/15) FINISHED                                                                                                                                                    
 => [internal] load build definition from Dockerfile                                                                                                                              0.0s
 => => transferring dockerfile: 514B                                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                                 0.0s
 => => transferring context: 2B                                                                                                                                                   0.0s
 => [internal] load metadata for docker.io/library/golang:1.21.3-bookworm                                                                                                         0.8s
 => CACHED [lb 1/6] FROM docker.io/library/golang:1.21.3-bookworm@sha256:5bafbbb109f02aaf6b41ddc19f54919773c3006f1cbda1599112603367642f0e                                         0.0s
 => [internal] load build context                                                                                                                                                 0.1s
 => => transferring context: 204.26kB                                                                                                                                             0.1s
 => [lb 2/6] RUN apt-get update                                                                                                                                                   3.1s
 => [lb 3/6] RUN apt-get -y install vim                                                                                                                                           4.4s
 => [lb 4/6] RUN cd /go/src && mkdir -p sctplb                                                                                                                                    0.7s 
 => [lb 5/6] COPY . /go/src/sctplb                                                                                                                                                0.2s 
 => [lb 6/6] RUN cd /go/src/sctplb && go mod tidy && go install                                                                                                                  19.9s 
 => [sctplb 1/4] WORKDIR /sdcore                                                                                                                                                  0.1s 
 => [sctplb 2/4] RUN mkdir -p /sdcore/bin                                                                                                                                         0.7s 
 => [sctplb 3/4] COPY --from=lb /go/bin/* /sdcore/bin/                                                                                                                            0.2s 
 => [sctplb 4/4] WORKDIR /sdcore                                                                                                                                                  0.1s 
 => exporting to image                                                                                                                                                            4.0s 
 => => exporting layers                                                                                                                                                           4.0s 
 => => writing image sha256:f9c0ace10fef293bf097930b2d102222abe13a2a44f00b85de32850b19b8964f                                                                                      0.0s
 => => naming to docker.io/library/sctplb:0.0.1-dev 
 ```

**Comment by gab-arrobo** [comment]
yest this please

**Comment by gab-arrobo** [comment]
test this please

**Pull request merged**